plugins/base/src/main/kotlin/renderers/DefaultRenderer.kt

```diff
@@ -139,7 +171,9 @@ abstract class DefaultRenderer<T>(
 
         root.children<ModulePageNode>().forEach { renderPackageList(it) }
 
-        renderPages(newRoot)
+        runBlocking {
+            renderPages(newRoot)
+        }
```

Comment:
I think you have forgotten about running insides of `runBlocking` block on some not-blocking dispatcher (like `Dispatchers.Default`), doing so you can make non-IO steps also run in parallel.